lib/omniauth/strategies/shopify.rb

```diff
@@ -94,7 +94,12 @@ def self.hmac_sign(encoded_params, secret)
       end
 
       def valid_permissions?(token)
-        token && (options[:per_user_permissions] == !token['associated_user'].nil?)
+        return false unless token
```

Comment:
How about:
```
return false unless token
return true if options[:per_user_permissions].nil?
options[:per_user_permissions] == !token['associated_user'].nil?
```
This seems more readable to me.

Reply:
I'm not sure that's quite the logic.

I think that would incorrectly return `true` if `token['associated_user']` is present but `options[:per_user_permissions]` is not.

Reply:
You're right, I've edited my suggestion to correct that

Reply:
And by now, it's very similar in terms of readability; I leave it up to you to decide which is better.

Reply:
So `blank?` and `present?` is Rails specific. When trying on the tests, it would error out complaining that these two methods are undefined. That's why we can't do that.